lib/modules/datasource/cpan/index.ts

```diff
@@ -107,6 +107,21 @@ export class CpanDatasource extends Datasource {
           changelogUrl: `https://metacpan.org/dist/${latestDistribution}/changes`,
           homepage: `https://metacpan.org/pod/${packageName}`,
         };
+
+        const latestVersionUrl = joinUrlParts(
+          registryUrl,
+          'v1/module/',
+          packageName,
+        );
+        const latestVersionRes = (
+          await this.http.getJson<MetaCpanApiFile>(latestVersionUrl)
+        ).body;
+        const latestVersion = latestVersionRes.module[0].version;
```

Comment:
This needs to be more "defensive" in case data is not as we expect

Reply:
it should use zod from beginning. also this causes an additional API request.

Reply:
It's necessary, unless it can be done with the new URL only?

Reply:
should be done for every **new** request. we can leave existing as is

Reply:
I meant that getting the latest tag from the new URL is necessary. Which means two requests per package is necessary unless the new URL data can obsolete the existing URL data